### PR TITLE
Changing C++ Compile Version

### PR DESCRIPTION
**Pull request by CursedRock17**
This is a small pull request directly related to issue #75 that will update the compiler from compiling with standard C++11 to C++17, which should resolve small errors on MacOS arm64 as a guarantee that `std::variant` works. This change would also require a similar update the CXX version in the performance_test_fixture package as mentioned in the above issue.

### CODE REVIEW

File: osrf_testing_tools_cpp/CMakeLists.txt [hunk 1]
```diff
@@ -6,7 +6,7 @@ set(osrf_testing_tools_cpp_VERSION 1.5.1)
 
 # Default to C++11
```
Comment (clalancette) [inline osrf_testing_tools_cpp/CMakeLists.txt:7]:
```suggestion
# Default to C++17
```

File: test_osrf_testing_tools_cpp/CMakeLists.txt [hunk 2]
```diff
@@ -4,7 +4,7 @@ project(test_osrf_testing_tools_cpp)
 
 # Default to C++11
```
Comment (clalancette) [inline test_osrf_testing_tools_cpp/CMakeLists.txt:5]:
```suggestion
# Default to C++17
```

File: osrf_testing_tools_cpp/src/memory_tools/CMakeLists.txt [hunk 3]
```diff
@@ -47,7 +47,7 @@ option(OSRF_TESTING_TOOLS_CPP_DISABLE_MEMORY_TOOLS
 # Memory tools doesn't work on Windows right now.
 # TODO(wjwwood): If on aarch64, so not set the preload environment variables, until fixed.
 # see: https://github.com/osrf/osrf_testing_tools_cpp/issues/3
-if(WIN32 OR CMAKE_SYSTEM_PROCESSOR STREQUAL "aarch64")
+if(WIN32 OR CMAKE_SYSTEM_PROCESSOR STREQUAL "aarch64" OR CMAKE_SYSTEM_PROCESSOR STREQUAL "aarm64")
```
Comment (clalancette) [inline osrf_testing_tools_cpp/src/memory_tools/CMakeLists.txt:50]: Why this change?

**Review by clalancette — CHANGES_REQUESTED** [review]
Thanks for the contribution!  I left a question and a couple of small issues in my review.

File: osrf_testing_tools_cpp/src/memory_tools/CMakeLists.txt (hunk 3 above)
Comment (mjcarroll) [reply osrf_testing_tools_cpp/src/memory_tools/CMakeLists.txt:50]: I think what CMake reports is pretty wildly variable based on a quick search: https://cmake.org/cmake/help/latest/variable/CMAKE_HOST_SYSTEM_PROCESSOR.html#variable:CMAKE_HOST_SYSTEM_PROCESSOR

I think if we have evidence that `uname -m` is returning something that isn't `aarch64`, it would be worth adding to this list.

Comment (CursedRock17) [reply osrf_testing_tools_cpp/src/memory_tools/CMakeLists.txt:50]: I assumed from that comments that this library should've followed suit in the path of a Unix machine so checking the system processor via the style guide made the most sense. To follow up though, I had accidently added an extra a to arm64 when I made the commit, the CMake file is triggered when `CMAKE_SYSTEM_PROCESSOR STREQUAL "arm64". But, the library will compile reguardless of the processor value, it just relies on a CMake_CXX_Standard update.

Comment (clalancette) [inline osrf_testing_tools_cpp/src/memory_tools/CMakeLists.txt]:
```suggestion
if(WIN32 OR CMAKE_SYSTEM_PROCESSOR STREQUAL "aarch64" OR CMAKE_SYSTEM_PROCESSOR STREQUAL "arm64")
```

**Review by clalancette — APPROVED** [review]

**Comment by clalancette** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/18251/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/12787/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/18926/)

**Comment by FantasticRhythm** [comment]
after I excuted 

> `colcon build `

followed some tutorials

I got some problem with it

> Summary: 99 packages finished [56.3s]
  1 package failed: performance_test_fixture
  9 packages aborted: ament_index_cpp launch_testing_ament_cmake mimick_vendor rosidl_parser rttest rviz_ogre_vendor tlsf tracetools urdf_parser_plugin
  31 packages had stderr output: ament_clang_format ament_clang_tidy ament_copyright ament_cppcheck ament_cpplint ament_flake8 ament_index_python ament_lint ament_lint_cmake ament_mypy ament_package ament_pclint ament_pep257 ament_pycodestyle ament_pyflakes ament_uncrustify ament_xmllint domain_coordinator iceoryx_utils launch launch_testing launch_xml launch_yaml osrf_pycommon osrf_testing_tools_cpp performance_test_fixture rosidl_cli rpyutils rviz_ogre_vendor tracetools_read tracetools_trace
  196 packages not processed

So, how could I change the c++ complie Version when I build the ROS2-galactic?
